### PR TITLE
WIP: 2 add translations and cache

### LINKED ISSUE #2: Add translations and cache
1) Add a new command line arg, that enables translations,
ex. --trans=en,jp <--- would translate English to Japanese 
feel free to name this or format this however, and it only does translations if this is on

2) Use OpenAI or something similar to do a translation 

3) Store the translation into a local file, and cache them, so we only make the calls once

4) Translations should show up in the video feed

### PR DESCRIPTION
**Pull request by mattkanwisher**

### CODE REVIEW

File: process_frames.py [hunk 1]
```diff
@@ -279,6 +285,71 @@ def run_ocr(self, image):
         else:
             print("No entry found")        
         return res, highlighted_image, annotations
+    
+    def call_openai_translation_api(self, content, target_lang):
```
Comment (mattkanwisher) [inline process_frames.py]: this should probably be broken to a separate file and combined with the gpt vision api calls

File: image_window.py [hunk 2]
```diff
@@ -118,7 +125,35 @@ def print_annotations(self, frame):
...
+                    try:
+                        cv2.rectangle(frame, top_left, bottom_right, (0, 0, 255), 2)  # BGR color format, red box
+                    except Exception as e:
+                        print(f"Weird: y1 must be greater than or equal to y0, but got {top_left} and {bottom_right} respectively. Swapping...")
+
+                    # Annotate text. Adjust the position if necessary.
+                    image = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+                    pil_image = Image.fromarray(image)
+
+                    font = ImageFont.truetype("C:\Windows\YuGothB.ttc", 35)
```
Comment (mattkanwisher) [inline image_window.py]: should be able to name a font and not use a path to it

File: ss.py [hunk 3]
```diff
@@ -100,7 +100,10 @@ def process_video(video_path):
             # Save to disk
             pil_image.save("window_capture.jpg")
             print(f"Frame at {frame_count//fps} seconds saved as debug.jpg")
-            frameProcessor.run_image(pil_image)
+            if translate:
+                frameProcessor.run_image(pil_image, translate)
```
Comment (mattkanwisher) [inline ss.py]: do you need this if statement?

File: process_frames.py [hunk 4]
```diff
@@ -303,9 +374,24 @@ def run_image(self, img):
...
+                    translation = self.run_translation(self.dialogues[last_played], translate)
+                    print("finished translation")
+                else:
+                    translation = ""
+
             self.previous_image = img
             self.last_annotations = annotations
+
+            if translate:
+                return last_played, self.previous_image, highlighted_image, annotations, translation
```
Comment (mattkanwisher) [inline process_frames.py]: this is super messy, returning different number of parameters from the function, always return the same number

File: process_frames.py (hunk 1 above)
Comment (ShaneKoNaung) [reply process_frames.py]: Done

File: ss.py (hunk 3 above)
Comment (ShaneKoNaung) [reply ss.py]: Removed redundant if statements.

**Comment by ShaneKoNaung** [comment]
Simple Translation Overlay is done. 

https://github.com/VoxlyHq/retro_voice/assets/26071811/ecbb8a50-cd55-4acf-a3eb-83ae1d944e32

**Comment by ShaneKoNaung** [comment]
Dialogue box background color and text color selection based on the current frame. 

Before
before_color_selection

After
after_color_selection

**Comment by ShaneKoNaung** [comment]
Simple Text warping by adding new line if the text length is longer than the dialogue box. 

Before
before_text_warping

After
after_text_warping

**Pull request merged**